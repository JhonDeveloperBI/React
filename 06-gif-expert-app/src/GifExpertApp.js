import React from 'react'

export const GifExpertApp = props => {
  return( 
    <>
    <h2>GifExpertApp</h2>
    <hr/>
    </>  
  )
}


