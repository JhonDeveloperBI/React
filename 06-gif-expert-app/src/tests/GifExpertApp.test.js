import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('test in component <GifExpertApp />', () => {
    
    test('should show correct component',() => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    })

    test('should show list categories', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

        
    })

    
})