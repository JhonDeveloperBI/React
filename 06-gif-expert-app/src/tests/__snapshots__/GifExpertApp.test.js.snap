// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in component <GifExpertApp /> should show correct component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`test in component <GifExpertApp /> should show list categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
